### ISSUE TITLE
iOS 8.1.0-rc: Xcode 'Archive' produces a corrupt archive

### ISSUE THREAD

**Issue by sampok** [react-native-mapbox-gl/maps#1097]
**Describe the bug**
iOS: Xcode 'Archive' produces a corrupt archive. Seems to be possibly related to iOS Mapbox SDK 5.8. Found a workaround, see below.

**To Reproduce**

- `npx react-native init RNMapboxTest`
- `cd RNMapboxTest`
- `yarn add @react-native-mapbox-gl/maps`
- `cd ios && pod install`
- Open ios/RNMapboxTest.xcworkspace in Xcode
- Set build to 'Any iOS Device' and set signing
- Build with Product > Archive. Wait.
- Build completes without errors, but it doesn't show up in Organizer
- When trying to open the build from `/Library/Developer/Xcode/Archives`, get error 'The archive may be corrupt or unreadable.'

**Expected behavior**
Expecting to be able to build an Archive for App Store submission.

**Screenshots**
<img width="532" alt="Screen Shot 2020-11-03 at 10 27 14 AM" src="https://user-images.githubusercontent.com/1516075/98010035-3e3e5500-1dc4-11eb-9c1a-4cbced06550d.png">

**Versions**
 - Platform: iOS
 - Xcode: 12.1
 - react-native-mapbox-gl Version 8.1.0-rc.9
 - React Native Version 0.63.3

**Possible reason & Workaround**

- When using react-native-mapbox-gl v 8.0.0, the archive build works. However I need the padding fixes in 8.1 rcs, so I continued looking.
- Noticed that 8.0.0 installed Mapbox iOS SDK 5.9.0, whereas 8.1.0-rc.9 installs 5.8.0. This is because in 8.0.0 podspec the SDK dependency is defined without a 'patch' number, like this: '\~> 5.7', leading to installation of the latest 5.X version. In 8.1.0-rc.9 it's defined with a patch: '\~> 5.8.0', installing version 5.8.X.
- Learned about `$ReactNativeMapboxGLIOSVersion` to explicitly set the iOS SDK version. However, it seems to apply also to some other dependencies, and if you set it to '\~> 5.9.0', @react-native-mapbox-gl-mapbox-static fails to install. However setting it to '\~> 5.8' leads to install of Mapbox iOS SDK 5.9.0 and latest 5.X for others. And now archive build works correctly!
- So the workaround for me was simply to add `$ReactNativeMapboxGLIOSVersion = '~> 5.8'` into the Podfile, and `pod install`.

**Comment by sampok** [comment]
The Archive build gets corrupted again if I set Mapbox iOS SDK to 6.X version (using `$ReactNativeMapboxGLIOSVersion`).

It's quite annoying as [5.9 has a symbols flickering while zooming out issue](https://github.com/mapbox/mapbox-gl-native/issues/16471), which 5.8 didn't have, and it's fixed in 6.X.

**Comment by sampok** [comment]
It's strange that the app builds fine with all Mapbox iOS SDK versions for Debug and Release, but with Archive it becomes corrupted (except with 5.9.0).

When I peek inside the generated .xcarchive file that Archive creates, I see that the 'corrupted ones' lack Info.plist file and SCMBlueprint folder with .xcscmblueprint file inside it. Also BCSymbolsMaps folder seems to have less items.

The Archive process completes without any errors. It does have a lot of warnings, but they look like kind of "normal" for React Native builds.

**Comment by SalihZinaty** [comment]
I can confirm that is also happening with rc6 and rc9 
i did not check the rest. 
i need rc6 for the padding fixing of the ios.

**Comment by cbrevik** [comment]
I've noticed the same issue on AppCenter. Tried pinning to `~> 5.8` but instead of a malformed archive, I now get the following error:
```
Print: Entry, "CFBundleIdentifier", Does Not Exist
##[error]Failed to automatically generate an `Export options` plist file to export an IPA. Please specify an `Export options` plist file in `Package options`.
```

If I downgrade to `8.1.0-rc.9` the build succeeds though.

**Comment by sampok** [comment]
> If I downgrade to `8.1.0-rc.4` the archiving succeeds though.

I guess a possible, even likely (?) explanation is that in rc4 the Mapbox SDK version was defined without a patch number, and now resolves to the highest minor version, i.e. 5.9.0. ([rc4 to rc5 diff](https://github.com/react-native-mapbox-gl/maps/compare/8.1.0-rc4...8.1.0.rc5#diff-5af36a915682810f0dfe340920e5fee52f98285598c502c454a4dd01cdac9ad2))

**Comment by chriscohoat** [comment]
I've been struggling with this too for a bit, and I found that downgrading cocoapods from version 1.10.0 to 1.9.3 fixed the corrupt archive issues above. The command I used:

```
gem install cocoapods -v '<= 1.9.3' -n /usr/local/bin
```

- react-native-mapbox-gl Version 8.1.0-rc.8
- React Native Version 0.63.2

**Comment by tomihuttunen** [comment]
We had similar issues with react-native-mapbox-gl 8.0.0 and the cocoapods downgrade to 1.9.3 resolved those as well. 1.10.0 did not create a valid archive.

**Comment by cbrevik** [comment]
I've tried downgrading our `Podfile.lock` to `1.9.3`, but I am unsure if the AppCenter VMs will use `1.9.3` in that case. They have `1.10.0` installed by default it seems. So unable to verify on my end.

**Comment by tomihuttunen** [comment]
Yes Appcenter does not respect the Podfile.lock version and will use whatever is installed to the runner machine. And the version was just upgraded to 1.10.0 and that is when my problems started.

I used this script to enforce to use the version defined in the Podfile.lock. Which is in general better because now I can be in control what is used.

https://github.com/microsoft/appcenter/blob/master/sample-build-scripts/react-native/compare-cocoapods/appcenter-post-clone.sh

**Comment by sampok** [comment]
> The Archive build gets corrupted again if I set Mapbox iOS SDK to 6.X version (using `$ReactNativeMapboxGLIOSVersion`).

Turns out that it's possible use Mapbox iOS SDK 6.X and avoid corruption with
```$ReactNativeMapboxGLIOSVersion = '~> 6.2.1'```

Previously I pinned to exact version `'6.2.1'` because the latest version 6.2.2 crashes in Xcode Release build on armv7 device. But it seems to work fine when using Archive and testing through TestFlight 🤷‍♂️..

**Comment by naftalibeder** [comment]
Downgrading to cocoapods `1.9.3` fixed this for me!

**Comment by westanvv** [comment]
> Downgrading to cocoapods `1.9.3` fixed this for me!

I have the same problem. Downgrading is working for me too

**Comment by sjusaitis** [comment]
Downgrading to Cocoapods `1.9.3` also fixed this issue for me. Has anyone manage to get this working with Cocoapods `1.10.0`?

**Comment by aprilzero** [comment]
Going back to 1.9.3 fixed it for us too

**Comment by patternleaf** [comment]
Quick note for other folks looking for this: I was able to get this working without overwriting my global cocoapods install by using a Gemfile with the following contents in the RN `ios` folder:

```
source 'https://rubygems.org'

# use cocoapods v 1.9.3 as work-around for mapbox issues when archiving
# https://github.com/react-native-mapbox-gl/maps/issues/1097
gem 'cocoapods', '~> 1.9.3'
```

Then instead of the usual `pod install` command, use `bundle exec pod install`.

**Comment by luskin** [comment]
There is a conflicting issue with the latest AppCenter packages though as they require `1.10.X` to archive. Anybody got this to work with any of the AppCenter packages on `4.0.X`?

**Comment by hugopretorius914** [comment]
Any news on this? I keep using 1.9.3 for this project and 1.10.1 for my other projects

**Comment by acecilia** [comment]
Was somebody able to report this issue in the cocoapods repository? I cant find any related issues 😞

**Comment by SanjanaTailor** [comment]
after running `pod install` command getting this **error ruby download_file_async': undefined method `encode' for URI:Module (NoMethodError)** due to that pod install not successfull. 

> Quick note for other folks looking for this: I was able to get this working without overwriting my global cocoapods install by using a Gemfile with the following contents in the RN `ios` folder:
> 
> ```
> source 'https://rubygems.org'
> 
> # use cocoapods v 1.9.3 as work-around for mapbox issues when archiving
> # https://github.com/react-native-mapbox-gl/maps/issues/1097
> gem 'cocoapods', '~> 1.9.3'
> ```
> 
> Then instead of the usual `pod install` command, use `bundle exec pod install`.

Tried this steps and after running commad `bundle exec pod install` getting this error **download_file_async': undefined method `encode' for URI:Module (NoMethodError)** under ruby folder, please help Thanks

**Comment by acaraty** [comment]
Using COCOAPODS: 1.10.1 and @react-native-mapbox-gl/maps": "8.1.0"

I had the exact same problem (archive succeed but actually malformed) Just open the xcarchive and you'll see there are missing parts (the plist for example).

The trick for me was to remove the copy dsym build phase of the @react-native-mapbox-gl-mapbox-static pod

This phase was causing build failure with some config and malformed archive with others, I don't really get the crux of the problem but may be it can help some of you...

**Comment by acecilia** [comment]
This is probably the issue, or related: https://github.com/CocoaPods/CocoaPods/issues/10385

**Comment by acaraty** [comment]
thanks @acecilia it really looks like the same problems, I'll give a try to your workaround

**Comment by SanjanaTailor** [comment]
> sing COCOAPODS: 1.10.1 and @react-native-mapbox-gl/maps": "8.1.0"



> Was somebody able to report this issue in the cocoapods repository? I cant find any related issues 😞

no not able to see facing same issue posted over here

**Comment by SanjanaTailor** [comment]
> after running `pod install` command getting this **error ruby download_file_async': undefined method `encode' for URI:Module (NoMethodError)** due to that pod install not successfull.
> 
> > Quick note for other folks looking for this: I was able to get this working without overwriting my global cocoapods install by using a Gemfile with the following contents in the RN `ios` folder:
> > ```
> > source 'https://rubygems.org'
> > 
> > # use cocoapods v 1.9.3 as work-around for mapbox issues when archiving
> > # https://github.com/react-native-mapbox-gl/maps/issues/1097
> > gem 'cocoapods', '~> 1.9.3'
> > ```
> > 
> > 
> > Then instead of the usual `pod install` command, use `bundle exec pod install`.
> 
> after following these steps and run the commad `bundle exec pod install`
> getting this error under ruby folder
> **download_file_async': undefined method `encode' for URI:Module (NoMethodError)** under ruby folder, please help Thanks
> 
> ruby version - 3.0.0 cocoapods - 1.9.3

**Comment by SanjanaTailor** [comment]
> the copy dsym build phase of the @react-native-mapbox-gl-mapbox-static pod

@acaraty - can you explain from where it is resolved - The trick for me was to remove the copy dsym build phase of the @react-native-mapbox-gl-mapbox-static pod ?

**Comment by AmalMenachery** [comment]
@SanjanaTailor Fixed the issue in starter project with @react-native-mapbox-gl/maps. Below are configuration and versions used:

- @react-native-mapbox-gl/maps - 8.1.0
- Ruby - 2.7.1
- cocoapods - 1.9.3
- Xcode 12.2

**Comment by acaraty** [comment]
@SanjanaTailor  

Removing the build phase fixed the malformed archive problem (in my case).

I'm not having errors when doing 'pod install' but I'm using an older ruby version (2.6), you might try to downgrade your ruby.

**Comment by SanjanaTailor** [comment]
@acaraty - yes tried to downgrade Ruby version from 3.3.0 to 2.7.1 and without any error IPA generated.

**Comment by SanjanaTailor** [comment]
> @SanjanaTailor Fixed the issue in starter project with @react-native-mapbox-gl/maps. Below are configuration and versions used:
> 
> * @react-native-mapbox-gl/maps - 8.1.0
> * Ruby - 2.7.1
> * cocoapods - 1.9.3
> * Xcode 12.2

Thank you @AmalMenachery Ruby version downgraded from 3.3.0 to 2.7.1 which resolved an issue of running this command below `bundle exec pod install` and after that IPA generated successfully

**Comment by SanjanaTailor** [comment]
> > after running `pod install` command getting this **error ruby download_file_async': undefined method `encode' for URI:Module (NoMethodError)** due to that pod install not successfull.
> > > Quick note for other folks looking for this: I was able to get this working without overwriting my global cocoapods install by using a Gemfile with the following contents in the RN `ios` folder:
> > > ```
> > > source 'https://rubygems.org'
> > > 
> > > # use cocoapods v 1.9.3 as work-around for mapbox issues when archiving
> > > # https://github.com/react-native-mapbox-gl/maps/issues/1097
> > > gem 'cocoapods', '~> 1.9.3'
> > > ```
> > > 
> > > 
> > > Then instead of the usual `pod install` command, use `bundle exec pod install`.
> > 
> > 
> > after following these steps and run the commad `bundle exec pod install`
> > getting this error under ruby folder
> > **download_file_async': undefined method `encode' for URI:Module (NoMethodError)** under ruby folder, please help Thanks
> > ruby version - 3.0.0 cocoapods - 1.9.3

@AmalMenachery's answer refer it or @acaraty answer helped to resolved an issue. Now archiving worked succesfully.

**Comment by ryderdonahue** [comment]
> @SanjanaTailor Fixed the issue in starter project with @react-native-mapbox-gl/maps. Below are configuration and versions used:
> 
> * @react-native-mapbox-gl/maps - 8.1.0
> * Ruby - 2.7.1
> * cocoapods - 1.9.3
> * Xcode 12.2

This was really helpful unblocking me yesterday! 

What is the longer term solution though? I imagine we can't just kick down our ruby/cocoapods version forever.

**Comment by acecilia** [comment]
@ryderdonahue Please try to read the full issue before asking. Its a bug in cocoapods: https://github.com/react-native-mapbox-gl/maps/issues/1097#issuecomment-770071623

**Comment by ryderdonahue** [comment]
> @ryderdonahue Please try to read the full issue before asking. Its probably a bug in cocoapods: [#1097 (comment)](https://github.com/react-native-mapbox-gl/maps/issues/1097#issuecomment-770071623)

My bad, I should have been clearer. I had seen the reference to the linked cocoapods issues, but my unfamiliarity with the underlying tech caused me to miss the connection between the actual problem vs a related one. 

Thanks for filing the issues on cocoapods @acecilia!

**Comment by BenjaminWatts** [comment]
Thanks for various contributions above. Not being a Swift/Ruby expert, I did struggle to follow your example repo @acecilia and integrate into my React Native project. 

Downgrading to v1.9.3 cocoapods fixed this issue for me, despite keeping **@react-native-mapbox-gl/maps** at v8.1.0.

Having had Cocoapods installed via Homebrew, I found it initially quite difficult to downgrade it (sudo errors), let alone manage multiple versions (so I could continue using newer Cocoapods on  other projects).

However, for my Mac, using [Ruby Version Manager](https://rvm.io/) did the trick. 

1. Install RVM: `\curl -sSL https://get.rvm.io | bash -s stable`
2. Follow the instructions printed at the end about updating the path to include the new Ruby you've installed (this will supplant the system version)
3. Install an older version of Ruby (2.7.0) did the trick for me through RVM: e.g. `rvm install 2.7.0` and `rvm use 2.7.0`
4. Install the old version of cocoapods using `pod install cocoapods -v 1.9.3`
5. To do the pod installation and force it to use the older version of cocoapods: `pod _1.9.3_ install`. This is handy as it will flag an error if it's not installed properly. I pop it a script within my `package.json`

One downside of doing this is that I've had to abandon Appcenter for building on iOS. I would guess you could get cloud building working on CircleCI, though I would warn that the installation (particularly of the pods) is pretty slow with the older version Cocoapods. 

Could anyone recommend a quick cloud building solution that would allow you to specify your version of Cocoapods?

Best of luck to everyone, and love Mapbox!

**Comment by RobertSasak** [comment]
@BenjaminWatts You can use _app-center-post-clone.sh_ to enforce correct cocoapod version. Create a new file in the root of your project directory. Check the pod version in _Podfile.lock_ as this is the one which will be used. It works and I did notice any special extra time on build. Be aware that AppCenter does not recognize _app-center-post-clone.sh_ unless you save new build settings.

This anser is just a summary of @tomihuttunen answer. Kudos to him!

**appcenter-post-clone.sh**
```sh
#!/usr/bin/env bash
CUR_COCOAPODS_VER=`sed -n -e 's/^COCOAPODS: \([0-9.]*\)/\1/p' ios/Podfile.lock`
ENV_COCOAPODS_VER=`pod --version`

# check if not the same version, reinstall cocoapods version to current project's
if [ $CUR_COCOAPODS_VER != $ENV_COCOAPODS_VER ];
then
    echo "Uninstalling all CocoaPods versions"
    sudo gem uninstall cocoapods --all --executables
    echo "Installing CocoaPods version $CUR_COCOAPODS_VER"
    sudo gem install cocoapods -v $CUR_COCOAPODS_VER
else 
    echo "CocoaPods version is suitable for the project"
fi;
```

**Comment by BenjaminWatts** [comment]
Cheers @RobertSasak . That looks super helpful - I will give it a go. Wasn't aware of the `--all --executables` options - suspect anyone trying to clean up local env (using my steps above) might find those helpful. PS - like norgeskart!

**Comment by acecilia** [comment]
Just a tip, the standard solution for installing cocoapods in a specific version (and the easier one) is to use a `Gemfile` and using it as explained here: https://github.com/react-native-mapbox-gl/maps/issues/1097#issuecomment-754757611

Installing cocoapods globally in a specific version is more complex and hacky :)

**Comment by ansarikhurshid786** [comment]
I have also downgrade. it's work fine thanks

**Comment by mrousavy** [comment]
Since React Native 0.64 (upcoming release) is not compatible with CocoaPods 1.9.3 anymore (because of Hermes afaik), simply downgrading to 1.9.3 is not, and never was a solution.

Has anyone found any information on what might be causing this?

**Comment by AmalMenachery** [comment]
@mrousavy Expo 40 is not compatible with Cocapods 1.9.3. Downgrading to 1.9.3 was a quick patch. 

This bug needs to be addressed for sure

**Comment by zholmes1** [comment]
Same issue here. I was not able to get a successful archive with Cocoapods 1.10. I had to downgrade to 1.9.3 to get a working archive.

**Comment by acecilia** [comment]
You can try the workarounds explained in the linked cocoapods issues I posted in https://github.com/react-native-mapbox-gl/maps/issues/1097#issuecomment-770071623 if you want to use cocoapods 1.10

**Comment by ebarahona** [comment]
> Quick note for other folks looking for this: I was able to get this working without overwriting my global cocoapods install by using a Gemfile with the following contents in the RN `ios` folder:
> 
> ```
> source 'https://rubygems.org'
> 
> # use cocoapods v 1.9.3 as work-around for mapbox issues when archiving
> # https://github.com/react-native-mapbox-gl/maps/issues/1097
> gem 'cocoapods', '~> 1.9.3'
> ```
> 
> Then instead of the usual `pod install` command, use `bundle exec pod install`.

Spent a good two days on this issue after an accidental computer restart triggered an Xcode update...I have tried everything, thank you!!!

**Comment by janicduplessis** [comment]
I was able to workaround this issue by forcing cocapods to use frameworks but only for the MapboxMobileEvents pod. This way I can still use static linking for other pods while fixing this and using cocoapods@1.10.1.

In my app Podfile:

```ruby
pre_install do |installer|
  fix_mapbox_events(installer)
end

def fix_mapbox_events(installer)
  installer.pod_targets.each do |pod|
    if pod.name == 'MapboxMobileEvents'
      def pod.build_type;
        Pod::BuildType.dynamic_framework
      end
    end
  end
end
```

Then update pods without the static linking fix:

```
REACT_NATIVE_MAPBOX_GL_USE_FRAMEWORKS=1 pod update
```

**Comment by cbrevik** [comment]
Awesome! That did the trick for me @janicduplessis. Thanks so much, this has been a thorn in my side.

Added the env-variable in our Podfile too: https://github.com/AtB-AS/mittatb-app/pull/935/files#diff-281ded35b124f5160d0a57e47e521ee6f49f233933e091a0d33f3fd42a74abc8R6 (so my team doesn't have to remember to set it locally / on CI).

Another issue I faced after was that `pod update` also broke our build because of a `libevent` conflict with Flipper. So a general `pod update` might trigger the same for others? Just `pod install` so it doesn't update all Pods worked for me 👌🏻 Though I see you've noticed that too: https://github.com/facebook/flipper/issues/1916

**Comment by anicol-weatherbug** [comment]
While this did fix our issue running the app locally using XCode, we find that during Archiving for the App Store that the MapboxMobileEvents.framework is missing from the .app folder. So the archive is created successfully, but crashes with a runtime error, since it cannot load the framework. 

Is anyone else still facing this issue?

**Comment by naftalibeder** [comment]
@anicol-weatherbug 

In your app target > `Build Phases` tab > `[CP] Embed Pods Frameworks` step, do you have `${BUILT_PRODUCTS_DIR}/MapboxMobileEvents/MapboxMobileEvents.framework` in your `Input Files` list?

**Comment by anicol-weatherbug** [comment]
@naftalibeder 
I do not see the inclusion of `MapboxMobileEvents.framework` in the `Embed Pods Frameworks` build step.

Podfile includes:
ENV['REACT_NATIVE_MAPBOX_GL_USE_FRAMEWORKS'] = '1'
$ReactNativeMapboxGLIOSVersion = '~> 6.3.3'

**Comment by janicduplessis** [comment]
@anicol-weatherbug If you go in xcode in the Pods project do you see MapboxMobileEvents as a framework? It will have this icon instead.

![image](https://user-images.githubusercontent.com/2677334/109543412-33ef4780-7a94-11eb-85ce-c17075b7f69e.png)

**Comment by anicol-weatherbug** [comment]
@janicduplessis For some reason it is still building as a static library, as I see` libMapboxMobileEvents.a `.

**Comment by mfazekas** [comment]
I've added a fix #1262

Note that it requires `Podfile` modification - see https://github.com/react-native-mapbox-gl/maps/pull/1262#issue-584279625

**Comment by ansarikhurshid786** [comment]
```
[!] An error occurred while processing the pre-install hook of the Podfile.

undefined method `pre_install' for nil:NilClass
```

**Comment by mfazekas** [comment]
> [!] An error occurred while processing the pre-install hook of the Podfile.
> 
> undefined method `pre_install' for nil:NilClass

Which version of @react-native-mapbox-gl/maps are you using. This is only fixed on `#master`so you should be using that.

**Comment by ansarikhurshid786** [comment]
react-native-mapbox-gl@8.1.0

**Comment by mfazekas** [comment]
> react-native-mapbox-gl@8.1.0

Please use `#master` or `8.2.0-beta1` I've just published beta1

**Comment by Alaa-Ben** [comment]
Same issue as @SebiVPS ! archive builds and available but error when uploading to app store: 

> ITMS-90562: Invalid Bundle - One or more dynamic libraries that are referenced by your app are not present in the dylib search path.

**Comment by mfazekas** [comment]
@Alaa-Ben, @SebiVPS what is the MPGL version used?!

**Comment by SebiVPS** [comment]
> @Alaa-Ben, @SebiVPS what is the MPGL version used?!

The latest version you mentioned above:
`"@react-native-mapbox-gl/maps": "^8.2.0-beta1"`

**Comment by mfazekas** [comment]
I meant if you have `$ReactNativeMapboxGLIOSVersion = '~> 6.3'`, or similar in your Podfile. If not please try with that too.

**Comment by janicduplessis** [comment]
@SebiVPS One way to verify the fix work is check if MapboxMobileEvents is actually being included as a framework per my comment here https://github.com/react-native-mapbox-gl/maps/issues/1097#issuecomment-788180364.

Maybe it doesn't work with certain cocoapod version? What version are you using?

**Comment by RobertSasak** [comment]
I would like to again bring attention to comment by @sampok 
https://github.com/react-native-mapbox-gl/maps/issues/1097#issuecomment-722498863
This one line resolved all the issues with corrupted archives while still using "@react-native-mapbox-gl/maps": "8.1.0",
```
$ReactNativeMapboxGLIOSVersion = '~> 6.2.1'
```
I can only confirm that 6.2.2 nor 6.3.3 version har still issue with corrupted archive.

**Comment by SebiVPS** [comment]
Thank you for your comments, i tried using `gem 'cocoapods', '~> 1.9.3'`
and `$ReactNativeMapboxGLIOSVersion = '~> 6.2.1'` in Podfile

Currently i get an error at `pod install`, looks like a temporarily issue at server side, i'll try again tomorrow:
```
[!] Error installing @react-native-mapbox-gl-mapbox-static
[!] /usr/bin/curl -f -L -o /var/folders/vc/5cybcdsd7zn7v1mbng40wx7h0000gp/T/d20210308-54315-1bh2bgn/file.zip https://api.mapbox.com/downloads/v2/mobile-maps/releases/ios/6.2.1/mapbox-ios-sdk-dynamic-with-events.zip --create-dirs --netrc-optional --retry 2 -A 'CocoaPods/1.9.3 cocoapods-downloader/1.4.0'

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 401 Unauthorized
```

**Comment by Alaa-Ben** [comment]
Hmm upgrading cocoapods to 1.10.2, upgrading mapboxgl to 8.2.0-beta1 and adding pre and post install methods in podfile solved this for me.

**Comment by RobertSasak** [comment]
@SebiVPS In order to use new Mapbox SDK you need to have a special password in .netrc. Please refer to Mapbox iOS SDK fo more info.
```sh
# Add Mapbox password to .netrc in order to download iOS SDK
echo -e "machine $NETRC_MAPBOX_MACHINE\nlogin $NETRC_MAPBOX_LOGIN\npassword $NETRC_MAPBOX_PASSWORD" > ~/.netrc
```

**Comment by janicduplessis** [comment]
I tested this patch on cocoapods 1.10.x so maybe it doesn't work on 1.9.x

**Comment by SebiVPS** [comment]
Thank you for pointing me in the right direction :) 
I got it to work with 
cocoapods v1.10.1, and
`"@react-native-mapbox-gl/maps": "^8.2.0-beta1"` and 
adding pre and post install methods in Podfile.

**Comment by bdtren** [comment]
> I've added a fix #1262
> 
> Note that it requires `Podfile` modification - see [#1262 (comment)](https://github.com/react-native-mapbox-gl/maps/pull/1262#issue-584279625)

Dear @mfazekas, I tried this solution. Archive and build success.
But I cannot upload the archived version to App Store, because of the `MapboxMobileEvents`

Error:
```
ERROR ITMS-90171: "Invalid Bundle Structure - The binary file 'App.app/Frameworks/MapboxMobileEvents.framework/MapboxMobileEvents' is not permitted. Your app can’t contain standalone executables or libraries, other than a valid CFBundleExecutable of supported bundles. Refer to the Bundle Programming Guide at https://developer.apple.com/go/?id=bundle-structure for information on the iOS app bundle structure."
```

**Comment by gregogalante** [comment]
I'm getting the same problem of @bdtren with 8.2.0-beta1 version and the pre_install and post_install tasks on my Podfile

**Comment by NathanNovak** [comment]
> Thank you for pointing me in the right direction :)
> I got it to work with
> cocoapods v1.10.1, and
> `"@react-native-mapbox-gl/maps": "^8.2.0-beta1"` and
> adding pre and post install methods in Podfile.

I get a corrupt archive and keep getting this error when trying to upgrade to 8.2.0-beta1:

npm ERR! Could not resolve dependency: npm ERR! peer react@"^16.6.1" from @react-native-mapbox-gl/maps@8.2.0-beta1 npm ERR! node_modules/@react-native-mapbox-gl/maps npm ERR! @react-native-mapbox-gl/maps@"8.2.0-beta1" from the root project
Can someone let me know how to solve this.

**Comment by abakhtiar** [comment]
I have the same corrupted archive  problem with "@react-native-mapbox-gl/maps": "^8.1.0-beta.1". Upgrading to latest versions also breaks my build for hooks dependancies. 
It is really a big mess in our apps! :(  
    "react": "16.8.6",
    "react-native": "0.60.5",
Anybody knows a way please let me know. 
Thanks

**Comment by NathanNovak** [comment]
@SebiVPS How did you get 8.2.0-beta1 to install on yours?

**Comment by SebiVPS** [comment]
@NathanNovak 
> @SebiVPS How did you get 8.2.0-beta1 to install on yours?

there is nothing special here, just add the specific version in your package.json and install. (No login needed)
But i still use `"react-native" : "0.63.4"`, maybe there is a problem with `0.64` and react v17 ?

**Comment by NathanNovak** [comment]
@SebiVPS Thats is the key difference right there.  React had a rather large changelog with 17.0.0.  

Does anyone know if using '--legacy-peer-deps' flag creates havoc?

**Comment by designervoid** [comment]
Same issue

**Comment by designervoid** [comment]
Fixed by upgrading from `"@react-native-mapbox-gl/maps": "^8.1.0",` to `"@react-native-mapbox-gl/maps": "^8.2.0-beta1",`.
Then `yarn install`, `npx pod-install`. And now I can see archived build in organizer.

**Comment by b3hz4d** [comment]
I did all the above solution with clean installation but I'm getting this error
```Info.plist: No such file or directory```
Any idea how to solve it?
Using cocoapods V1.10.1 & Xcode 12.4

**Comment by ghadeeraqraa1992** [comment]
> I did all the above solution with clean installation but I'm getting this error
> `Info.plist: No such file or directory`
> Any idea how to solve it?
> Using cocoapods V1.10.1 & Xcode 12.4

I have the same issue 
did you solve it ?

**Comment by Hless** [comment]
@ghadeeraqraa1992 Similar thing happend to me out of the blue, you may want to try the following thread: https://github.com/mapbox/mapbox-gl-native-ios/issues/555

**Comment by ghadeeraqraa1992** [comment]
> @ghadeeraqraa1992 Similar thing happend to me out of the blue, you may want to try the following thread: [mapbox/mapbox-gl-native-ios#555](https://github.com/mapbox/mapbox-gl-native-ios/issues/555)
> 
> Edit: ahhh nevermind, you already found that one

did you find a solution ?
